redis/pool.go

```diff
@@ -376,6 +359,52 @@ func (p *Pool) lazyInit() {
 	p.mu.Unlock()
 }
 
+// waitVacantConn waits until at least one vacant connection gets spotted in the pool.
+// If `ctx` expires before that, an error is returned.
```

Comment:
```suggestion
// If ctx expires before that, an error is returned.
```
Doc strings shouldn't use markdown syntax.